x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/WaterfallItem.tsx

```diff
@@ -17,29 +19,54 @@ import {
 } from '../../../../../../style/variables';
 import { IWaterfallItem } from './waterfall_helpers/waterfall_helpers';
 
-const ItemBar = styled.div`
+interface ItemBarProps {
+  type: 'transaction' | 'span';
+  left: number;
+  width: number;
+  color: string;
+}
+
+const ItemBar = styled<ItemBarProps, any>('div')`
+  box-sizing: border-box;
   position: relative;
-  height: ${unit}px;
+  height: ${props => (props.type === 'transaction' ? unit : unit)}px;
+  left: ${props => `${props.left}%`};
+  width: ${props => props.width}%;
+  min-width: '2px';
+  background-color: ${props => props.color};
 `;
-const ItemLabel = styled.div`
+
+const SpanLabel = styled<{ left: number }, any>('div')`
   white-space: nowrap;
   position: relative;
-  direction: rtl;
+  left: ${props => `${props.left}%`};
+  width: ${props => `${100 - props.left}%`};
   text-align: left;
   margin: ${px(units.quarter)} 0 0;
   font-family: ${fontFamilyCode};
+  font-size: ${fontSizes.tiny};
+`;
+
+const TransactionLabel = styled(SpanLabel)`
+  font-weight: 600;
+  font-family: ${fontFamily};
   font-size: ${fontSizes.small};
+  line-height: 1.5;
 `;
 
-const Container = styled<
-  { timelineMargins: TimelineMargins; isSelected: boolean },
-  'div'
->('div')`
+interface IContainerProps {
+  item: IWaterfallItem;
+  timelineMargins: ITimelineMargins;
+  isSelected: boolean;
+}
```

Comment:
You think you can do:
```ts
interface IContainerProps = Pick< IWaterfallItemProps['transaction'], 'item' | 'timelineMargins', 'isSelected'>
```

Not sure it's better (harder to read), but perhaps it's nice not to duplicate the interface. What you think?